openlibrary/plugins/openlibrary/code.py

```diff
@@ -205,11 +205,13 @@ def sampleload(filename='sampledump.txt.gz'):
     if filename.endswith('.gz'):
         import gzip
 
-        f = gzip.open(filename)
+        with gzip.open(filename) as file:
+            pass
     else:
-        f = open(filename)
+        with open(filename) as file:
+            pass
 
-    queries = [json.loads(line) for line in f]
+    queries = [json.loads(line) for line in file]
```

Comment:
This will likely need to be something like:

```
import gzip

with (gzip.open(filename) if filename.endswith('.gz') else open(filename)) as file:
    queries = [json.loads(line) for line in file]
```

Once we're out of the `with` block, the `file` will no longer be accessible otherwise!